### ISSUE TITLE
Change event state not enabling events

### ISSUE THREAD

**Issue by LostLinguine** [Nain57/Smart-AutoClicker#359]
I have 4 events.. 1.) Aura 1
2.) Confirm attack buff. 3.) Aura 2. 4.) Confirm attack buff 2.. when what aura 1 is detected and attack buff is detected it'll disable aura 1 enable arua 2 then disable itself, aura 2 is supposed to enable the second attack buff event but it doesn't for some reason and I concluded that somehow event state can't change an events state when it's already disabled( I enabled the 2nd confirm attack buff and it detects fine but it can't be enabled for some reason and it's very annoying to deal with.

I can't explain how to reproduce it since it's a mystery to me as to why I can't enable an event and have it work normally

I expected it to work as normal( works fine on 2.0.2 but not on 2.3.6) it's supposed to buff an attack but since it's rng I need the clicker to know when a buff lands so it can consistently buff itself in a raid for automation but the confirmation event isn't working like it used to.

[auto-raid_scissors.zip](https://github.com/Nain57/Smart-AutoClicker/files/13620350/auto-raid_scissors.zip)

-The phone used: tcl ax2 A507DL
-Android 10
-smart clicker version 2.3.6

**Comment by AardCrowley** [comment]
I am experiencing this as well. I have a set of conditions that enable/disable progressively. It seems iffy whether something is enabled or not, and oftentimes I find that the autoclicker seems to 'hang'. Unfortunately, the debug option is barebones and does not offer any insight on whether something was enabled/disabled. A verbose debug log would be amazing, and could provide information such as:

Event MyStart triggered Condition MyImage at position 100, 258
MyImage clicked at position 100, 258
Event ToggleMe enabled
Event ToggleMe triggered Condition MyPicture at position 382, 2882
MyPicture swiped from position 500, 2882 to 200, 2882
Event ToggleMe2 enabled
Event ToggleMe disabled

and so forth. I think that would provide deeper debug logs for anyone to determine what exactly is going on. Just food for thought, anyway. But this bug is definitely annoying.

**Comment by hsantanna** [comment]
Apparently disabling and then enabling events modifies the priority of the events, so that the order explicitly defined in the event list is no longer respected. 
That's just a suspicion of mine, I'm not sure how to actually confirm beyond the preliminary tests.

**Comment by Nain57** [comment]
Hi.
I'm currently reworking this feature to make it easier to use, i will take a look at the problem while i'm it.

Sorry for the inconvenience

**Comment by Nain57** [comment]
A little update about this issue:

The following exception is raised by the system when rotating the device on Android 14 once i am recording the screen:
```
java.lang.SecurityException: Don't re-use the resultData to retrieve the same projection instance, and don't use a token that has timed out. Don't take multiple captures by invoking MediaProjection#createVirtualDisplay multiple times on the same instance.
    at android.os.Parcel.createExceptionOrNull(Parcel.java:3087)
    at android.os.Parcel.createException(Parcel.java:3071)
    at android.os.Parcel.readException(Parcel.java:3054)
    at android.os.Parcel.readException(Parcel.java:2996)
    at android.hardware.display.IDisplayManager$Stub$Proxy.createVirtualDisplay(IDisplayManager.java:1450)
    at android.hardware.display.DisplayManagerGlobal.createVirtualDisplay(DisplayManagerGlobal.java:657)
    at android.hardware.display.DisplayManager.createVirtualDisplay(DisplayManager.java:1169)
    at android.media.projection.MediaProjection.createVirtualDisplay(MediaProjection.java:247)
    at android.media.projection.MediaProjection.createVirtualDisplay(MediaProjection.java:216)
    at com.buzbuz.smartautoclicker.core.display.DisplayRecorder.startScreenRecord(DisplayRecorder.kt:156)
    at com.buzbuz.smartautoclicker.core.processing.data.DetectorEngine$onOrientationChanged$1.invokeSuspend(DetectorEngine.kt:216)
    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
    at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
    at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:115)
    at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:103)
    at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
    at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
    at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
    at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
```

It seems like a protection has been added to avoid asking the screencast permission only once and saving it to use it whenever the apps want. This is really a good addition to Android from a user security perspective, but it seems like they accidentaly locked apps like Smart AutoClicker to using the permission correclty and just change the projection size when they need it (This how my app works, it basically screencast the screen in a buffer)

I'm currently trying to work around this new limitation without requesting the screencast permission for each rotation, but this might be tricky to fix. I will keep you updated !

**Comment by hsantanna** [comment]
In cases of large scenarios, with many events that disable and enable each other, it is hard to debug and know which event changed the other. It would be nice to have an easy way to list which events affect the state of a specific event.

**Comment by nucspl** [comment]
Hi, I just checked beta04 to see if this issue has been addressed from 2.4.2, and it seems that I still can't enable/toggle a disabled event, despite the 'observer'/toggler detecting its condition and working with other actions (click, swipe, etc.).

**Comment by Nain57** [comment]
Closed: This feature have drastically changed since this bug was open. If you still encounter an issue, please open a new bug ticket.
